packages/E2ETest/windows/TreeDumpLibrary/TreeDumpControlViewManager.cs

```diff
@@ -0,0 +1,191 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Diagnostics;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Controls;
+
+using Microsoft.ReactNative.Managed;
+using Microsoft.ReactNative.Bridge;
+using Windows.UI.Xaml;
+using Windows.UI.ViewManagement;
+using System.Threading.Tasks;
+using System;
+using System.IO;
+using Windows.Storage;
+
+namespace TreeDumpLibrary
+{
+    internal class TreeDumpControlViewManager : AttributedViewManager<TextBlock>
+    {
+        public override string Name => "TreeDumpControl";
+        public TreeDumpControlViewManager(IReactContext reactContext) : base(reactContext) { }
+
+        public override FrameworkElement CreateView()
+        {
+            m_textBlock = new TextBlock();
+            m_textBlock.TextWrapping = TextWrapping.Wrap;
+            m_textBlock.IsTextSelectionEnabled = false;
+            m_textBlock.LayoutUpdated += async (source, e) =>
+            {
+                var bounds = ApplicationView.GetForCurrentView().VisibleBounds;
+                if (bounds.Width != 1024 || bounds.Height != 768)
+                {
+                    // Dump disabled when window size is not 1024x768!
+                    UpdateResult(false /*matchDump*/, "Window has been resized, dump comparison is only valid at default launch size: 1024x768!");
+                }
+                else
+                {
+                    await MatchTreeDumpFromLayoutUpdateAsync();
```

Comment:
>MatchTreeDumpFromLayoutUpdateAsync [](start = 26, length = 34)

Will it introduce unnecessary layout cycle? For example. View is ready -> Layout updated -> not matched -> Textblocked updated -> Layout updated ->

Reply:
It will cause layout once when text changes from Failed to Passed. If there are additional failure, I don't update TextBlock.Text. User has to click on it to view all saved error messages.

---
In reply to: [356297384](https://github.com/microsoft/react-native-windows/pull/3754#discussion_r356297384) [](ancestors = 356297384)